frontend/src/components/PrerequisiteTree/PrerequisiteTree.tsx

```diff
@@ -113,21 +113,25 @@ const PrerequisiteTree = ({ courseCode }: Props) => {
 
       // render graph
       if (!graph) {
-        generateTreeGraph(graphData);
+        if (graphData.children.length === 0) {
+          <p>No prerequisite visualisation is needed for this course</p>;
+        } else {
+          generateTreeGraph(graphData);
+        }
       } else {
         // NOTE: This is for hot reloading in development as new graph will instantiate every time
         updateTreeGraph(graphData);
       }
-
       setLoading(false);
     };
-
     if (courseCode) setupGraph(courseCode);
   }, [courseCode, degree, dispatch, graph, planner]);
 
   return (
     <S.PrereqTreeContainer ref={ref} height={calcHeight(coursesRequires, courseUnlocks)}>
       {loading && <Spinner text="Loading tree..." />}
+      {!loading && (!graph?.getNodes)
```

Comment:
nitpick: do we need the brackets over `!graph?.getNodes`? Might aswell remove it to be consistent with style

Reply:
true

Reply:
when you mean revert i still need the second if condition because that determines if there are any children:

`if (!graph && graphData.children.length !== 0) {
        generateTreeGraph(graphData);
      } else {
        // NOTE: This is for hot reloading in development as new graph will instantiate every time
        updateTreeGraph(graphData);
      }`

Reply:
Oh wait I just realised it should be `{!loading && graph?.getNodes() <= 1 && ...` rather than `{!loading && (!graph?.getNodes) ...`. Forgot the brackets haha since getNodes is a function/method.

For the revert, I think it doesn't matter (or atlease I dont think). The graph should generate with 1 children (the course node (i.e. if the tree belongs to comp1511, it will show the comp1511 node only if there are no children) and hence the `graph.getNodes()` will return only 1 which you can use to compare.

Reply:
Oooo wait nvm, dont mind what I said, i Think I get what you mean. Dont need to revert it if it works on your end. This should just be fine then? `if (!graph && graphData.children.length !== 0) { generateTreeGraph(graphData); } else { // NOTE: This is for hot reloading in development as new graph will instantiate every time updateTreeGraph(graphData); }`

Reply:
oh ok sweet!